python/ray/experimental/tf_utils.py

```diff
@@ -195,3 +200,353 @@ def set_weights(self, new_weights):
                 for (name, value) in new_weights.items()
                 if name in self.placeholders
             })
+
+
+DEBUG = 0
+def logger_debug_info(txt):
+    if DEBUG:
+        logger.info(txt)
+
+banner_half = "".join(["#"] * 10)
+
+def tf_differentiable(num_return_vals):
+    def tf_differentiable_helper(method):
+        #TODO(vsatish): Can we automatically determine 
+        #that a method is differentiable at runtime based on inputs?
+
+#        def differentiable_method(self, identifier, forward_pass, 
+#                                  persistent_tape, dys,
+#                                  arg_types, kwarg_types, *args, 
+#                                  **kwargs):
+        def differentiable_method(self, identifier, forward_pass, 
```

Comment:
This needs an informative doc string explaining the concepts and design decisions as well as the meanings of the arguments.

For doc strings throughout, please exactly follow the style in http://www.sphinx-doc.org/en/1.5/ext/example_google.html